src/ngrest/Config.php

```diff
@@ -1,660 +1,661 @@
-    }
-
-    private $_plugins;
-    
-    /**
-     * Get all plugins.
-     *
-     * @return array
-     */
-    public function getPlugins()
-    {
-        if ($this->_plugins === null) {
-            $plugins = [];
-            foreach ($this->getConfig() as $pointer => $fields) {
-                if (is_array($fields)) {
-                    foreach ($fields as $field) {
-                        if (isset($field['type'])) {
-                            $fieldName = $field['name'];
-                            if (!array_key_exists($fieldName, $plugins)) {
-                                $plugins[$fieldName] = $field;
-                            }
-                        }
-                    }
-                }
-            }
-            $this->_plugins = $plugins;
-        }
-
-        return $this->_plugins;
-    }
-
-    private $_extraFields;
-    
-    /**
-     * Get all extra fields.
-     *
-     * @return array
-     */
-    public function getExtraFields()
-    {
-        if ($this->_extraFields === null) {
-            $extraFields = [];
-            foreach ($this->getConfig() as $pointer => $fields) {
-                if (is_array($fields)) {
-                    foreach ($fields as $field) {
-                        if (isset($field['extraField']) && $field['extraField']) {
-                            if (!array_key_exists($field['name'], $extraFields)) {
-                                $extraFields[] = $field['name'];
-                            }
-                        }
-                    }
-                }
-            }
-            $this->_extraFields = $extraFields;
-        }
-
-        return $this->_extraFields;
-    }
-
-    /**
-     * @inheritdoc
-     */
-    public function getPointerExtraFields($pointer)
-    {
-        $extraFields = [];
-        foreach ($this->getPointer($pointer, []) as $field) {
-            if (isset($field['extraField']) && $field['extraField']) {
-                $extraFields[] = $field['name'];
-            }
-        }
-        return $extraFields;
-    }
-
-    /**
-     * @inheritdoc
-     */
-    public function onFinish()
-    {
-        foreach ($this->getPrimaryKey() as $pk) {
-            if (!$this->hasField('list', $pk)) {
-                $alias = $pk;
-                
-                if (array_key_exists($alias, $this->_attributeLabels)) {
-                    $alias = $this->_attributeLabels[$alias];
-                } elseif (strtolower($alias) == 'id') {
-                    $alias = Module::t('model_pk_id'); // use default translation for IDs if not label is given
-                }
-                
-                $this->addField('list', $pk, [
-                    'name' => $pk,
-                    'alias' => $alias,
-                    'type' => [
-                        'class' => 'luya\admin\ngrest\plugins\Number',
-                        'args' => [],
-                    ],
-                ]);
-            }
-        }
-    }
-}
+<?php

```

Comment:
@rochdi80tn can you explain me why every line has changes, but you have NOT changed anything? This looks strange.

Reply:
Have you set tabs as tabs instead of 4 spaces maybe?

Reply:
Not sure.. I am using a linux shell WSL mode for git commits.. 
Hadn't messed up with tabs and spaces..
I'll double check this odd behavior